### PR TITLE
feat(score): added score page

### PR DESCRIPTION
**Pull request by pjonatansr**
# Score
That PR is part of #270 

## Score Page is under construction :)
Now you can see your score on /score route

- [x] created page to show score
- [x] changed score calculation to avoid using mutability
- [x] added token to session

### CODE REVIEW

**Comment by pjonatansr** [comment]
I just saw my twitter username hard-coded. 
I will fix it soon.

File: apps/web/.env.example [hunk 1]
```diff
@@ -1,4 +1,6 @@
+NEXTAUTH_SECRET= # Linux: `openssl rand -hex 32` or go to https://generate-secret.now.sh/32
 TWITTER_BEARER_TOKEN=
 TWITTER_CLIENT_ID=
 TWITTER_CLIENT_SECRET=
-NEXTAUTH_SECRET= # Linux: `openssl rand -hex 32` or go to https://generate-secret.now.sh/32
\ No newline at end of file
+TWITTER_PROFILE_ID=#2313873457
+TWITTER_PROFILE_USER=#sseraphini
```
Comment (sibelius) [inline apps/web/.env.example:6]: why twitter profile id and user is required?

File: apps/web/src/components/score/Score.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    like_count: 0,
+    quote_count: 0,
+    total: 0,
+  });
+
+  useEffect(() => {
+    if (session) {
+      const updateScore = async (session: any) => {
+        const body = new URLSearchParams({
+          providerAccountId: session?.id,
```
Comment (sibelius) [inline apps/web/src/components/score/Score.tsx:24]: can't we consume this data in the backend?

can't we use `useSession` in the backend?

File: apps/web/src/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -1,43 +1,33 @@
...
+        token.access_token = account.access_token;
+      }
+      return token;
+    },
   },
   providers: [
     TwitterProvider({
       clientId: config.TWITTER_CLIENT_ID,
       clientSecret: config.TWITTER_CLIENT_SECRET,
       version: '2.0',
```
Comment (sibelius) [inline apps/web/src/pages/api/auth/[...nextauth].ts:30]: no more userinfo ?

File: apps/web/src/pages/api/userScore.ts [hunk 4]
```diff
@@ -1,69 +1,87 @@
...
-    public_metrics: {
-      retweet_count: number;
-      reply_count: number;
-      like_count: number;
-      quote_count: number;
-    };
-  }
-
-  const scoreDetail = {
+export default async function handler(
```
Comment (sibelius) [inline apps/web/src/pages/api/userScore.ts:6]: userScoreHandler is a better name

File: apps/web/src/pages/api/userScore.ts [hunk 5]
```diff
@@ -1,69 +1,87 @@
...
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  const { body } = req;
+  const { providerAccountId, access_token, username } = body;
```
Comment (sibelius) [inline apps/web/src/pages/api/userScore.ts:11]: grab this data directly from backend

File: apps/web/src/pages/score.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      <Flex
+        flex={1}
+        h={'100vh'}
+        w={'100%'}
+        alignItems="center"
+        justifyContent="center"
+        flexDirection="column"
+        pb="10px"
+      >
+        <SessionProvider session={session}>
```
Comment (sibelius) [inline apps/web/src/pages/score.tsx:31]: should we move this inside _app or _document?

**Review by sibelius — COMMENTED** [review]
needs more polishing

File: apps/web/.env.example (hunk 1 above)
Comment (pjonatansr) [reply apps/web/.env.example:6]: cc sseraphini as a service

Comment (sibelius) [reply apps/web/.env.example:6]: sounds good, we need to document this somehow

Comment (pjonatansr) [reply apps/web/.env.example:6]: I think we can avoid repeat your username in the code, so if in the future we build that for any people runs your instance, we'll not need to do refactor all the codebase.
See:
image

File: apps/web/src/pages/api/auth/[...nextauth].ts (hunk 3 above)
Comment (pjonatansr) [reply apps/web/src/pages/api/auth/[...nextauth].ts:30]: It has a default, you can see here: https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/twitter.ts
image

I think we can put userInfo again if we need to use it for something

File: apps/web/src/pages/api/userScore.ts (hunk 4 above)
Comment (pjonatansr) [reply apps/web/src/pages/api/userScore.ts:6]: Do I change the handler of tweets.ts to tweetsHandler ?
image

Comment (sibelius) [reply apps/web/src/pages/api/userScore.ts:6]: we need to normalize

**Pull request merged**